localstack/services/cloudformation/engine/template_deployer.py

```diff
@@ -97,9 +97,16 @@ def get_attr_from_model_instance(
 
     attribute_candidate = properties.get(attribute_name)
     if "." in attribute_name:
+        # in case we explicitly add a property with a dot, e.g. resource["Properties"]["Endpoint.Port"]
+        # this is only really used for legacy cases since this isn't supported in the propertiers part of the schema
         if attribute_candidate:
-            # in case we explicitly add a property with a dot, e.g. resource["Properties"]["Endpoint.Port"]
             return attribute_candidate
+
+        # some resources (e.g. ElastiCache) have their readOnly attributes defined as Aa.Bb but the property is named AaBb
+        if attribute_candidate := properties.get(attribute_name.replace(".", "")):
+            return attribute_candidate
+
```

Comment:
Is this something better fixed in the provider? I can see centralising the logic here means that provider writers do not have to be aware of CFn internals, but this might become difficult to maintain in the future.

Reply:
In general, like the quirks, I agree we should move  them to the provider in the future where they would just be ignored on AWS. For now handling schema inconsistencies is done in the Service to stay consistent with the rest of the quirks.